src/views/ToDoList/ToDoList.test.tsx

```diff
@@ -0,0 +1,262 @@
+      id: 1,
+      userId: 1,
+      completed: false,
+    };
+
+    (toDosApi.fetchToDos as Mock).mockReturnValue(new Promise(() => null));
+  });
+
+  describe('when the component is mounted', () => {
+    describe('and the Api call is successful', () => {
+      beforeEach(() => {
+        (toDosApi.fetchToDos as Mock).mockResolvedValue(returnedTasksList);
+      });
+      it('should display tasks and not display error', async () => {
+        const toDoList = render(<ToDoList />);
+
+        const firstTasksParagraph = await toDoList.findByText(task1.title, {
+          selector: 'p',
+        });
+        const secondTasksParagraph = await toDoList.findByText(task2.title, {
+          selector: 'p',
+        });
+        const thirdTasksParagraph = await toDoList.findByText(task3.title, {
+          selector: 'p',
+        });
+        const fourthTasksParagraph = await toDoList.findByText(task4.title, {
+          selector: 'p',
+        });
+        const errorParagraph = toDoList.queryByTestId('comment-error');
+
+        expect(firstTasksParagraph).toBeDefined();
+        expect(secondTasksParagraph).toBeDefined();
+        expect(thirdTasksParagraph).toBeDefined();
+        expect(fourthTasksParagraph).toBeDefined();
+        expect(errorParagraph).toBeNull();
+      });
+    });
+    describe('and the Api call is unsuccessful', () => {
+      beforeEach(() => {
+        (toDosApi.fetchToDos as Mock).mockRejectedValue(null);
+      });
+      it('should display error and not display tasks', async () => {
+        const toDoList = render(<ToDoList />);
+
+        const errorParagraph = await toDoList.findByText(
+          'Failed to catch tasks list',
+          { selector: 'p' },
+        );
+        const firstTaskParagraph = toDoList.queryByText(task1.title, {
+          selector: 'p',
+        });
+
+        expect(errorParagraph).toBeDefined();
+        expect(firstTaskParagraph).toBeNull();
+      });
+    });
+  });
+  describe('when user types the title', () => {
+    describe('and clicks the submit button', () => {
+      beforeEach(() => {
+        (toDosApi.fetchToDos as Mock).mockResolvedValue(returnedTasksList);
+        (toDosApi.createNewTask as Mock).mockReturnValue(
+          new Promise(() => null),
+        );
+      });
+      it('should make attempt to create the POST request', () => {
+        const toDoList = render(<ToDoList />);
+
+        const titleInput = toDoList.getByPlaceholderText('title');
+
+        fireEvent.input(titleInput, { target: { value: newTaskTitle } });
+
+        const button = toDoList.getByRole('button');
+
+        fireEvent.click(button);
+
+        expect(toDosApi.createNewTask).toHaveBeenCalledWith(newTaskMock);
+      });
+      describe('if the Api call is successful', () => {
+        beforeEach(() => {
+          (toDosApi.createNewTask as Mock).mockResolvedValue(newTaskMock);
+        });
+        it('should create new task and not throw error', async () => {
```

Comment:
`not throw error` is a bit vague. Here you are checking if the error is **rendered**

Also,
- displaying the created task
- not displaying an error
are two separate test cases and should have two separate its